x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -299,7 +299,18 @@ void refresh(PersistentTasksCustomMetaData persistentTasks, ActionListener<Void>
                 }
                 fullRefreshCompletionListeners.clear();
             }
-        }, onCompletion::onFailure);
+        },
+        e -> {
+            synchronized (fullRefreshCompletionListeners) {
+                assert fullRefreshCompletionListeners.isEmpty() == false;
+                for (ActionListener<Void> listener : fullRefreshCompletionListeners) {
+                    listener.onFailure(e);
```

Comment:
I was going to comment on how we don't signal `onCompletion` any more, but then I saw line 286. This is definitely a sneaky bug.

Reply:
Yes, as well as making it impossible to retry the bug also meant that only 1 of the queuing listeners got notified of failures.  The others didn't receive any notification at all.